lib/src/domain/location/get_location_use_case.dart

```diff
@@ -0,0 +1,33 @@
+import 'package:allthenews/src/domain/location/location.dart';
+import 'package:allthenews/src/domain/location/location_exception.dart';
+import 'package:allthenews/src/domain/location/location_resolver.dart';
+
+class GetLocationUseCase {
+  final LocationResolver _locationResolver;
+
+  GetLocationUseCase(this._locationResolver);
+
+  Future<Location> call() async {
+    final isServiceEnabled = await _locationResolver.isLocationServiceEnabled();
+    if (!isServiceEnabled) {
+      return Future.error(LocationServiceDisabledException());
+    }
+
+    if (await _locationResolver.isPermissionDeniedForever()) {
+      return Future.error(PermissionDeniedForeverException());
+    }
+
+    if (await _locationResolver.isPermissionDenied()) {
+      await _locationResolver.requestPermission();
+    }
+
+    if (await _locationResolver.isPermissionDeniedForever()) {
+      return Future.error(PermissionDeniedForeverException());
+    }
+    if (await _locationResolver.isPermissionDenied()) {
+      return Future.error(PermissionDeniedException());
+    }
+
+    return _locationResolver.getCurrentLocation();
```

Comment:
myślę, że możemy polegać na błędach zwracanych z tej metody przy nieudanym pobraniu danych skoro obsługuje to biblioteka zamiast pisać swoje ify